src/Phinx/Migration/Manager.php

```diff
@@ -392,13 +392,16 @@ public function rollback($environment, $version = null, $force = false)
             // Get the migration before the last run migration
             $prev = count($versions) - 2;
             $version =  $prev < 0 ? 0 : $versions[$prev];
+        } elseif ($version === 'all' || $version == 0) {
+            $version = 0;
         } else {
-            // Get the first migration number
-            $first = $versions[0];
+            // Array of migration names
+            $migrationNames = array_map(function ($item) { return $item['migration_name']; }, $versionLog);
 
-            // If the target version is before the first migration, revert all migrations
-            if ($version < $first) {
-                $version = 0;
+            // Try to find a migration id with the supplied version name
+            $found = array_search($version, $migrationNames);
+            if ($found !== false) {
+                $version = $found;
             }
         }
```

Comment:
Nice catch. I'll update the tests and add that change in. I'm a little busy for the next few days but I'll see how I go by the weekend